### PR TITLE
Implement game factory

### PR DESCRIPTION
**Pull request by Katerina-codes**
- Refactor code further by using a game factory to remove duplication and make the code more concise.

@MollieS @Maikon @jsuchy @felipesere

### CODE REVIEW

File: src/main/game/Language/GameFactory.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        this.inputOutput = inputOutput;
+    }
+
+    public List<Player> getPlayerTypes(String playerTypes) {
+        Player[] humanVersusHuman = {new Human(inputOutput), new Human(inputOutput)};
+        Player[] humanVersusComputer = {new Human(inputOutput), new Computer()};
+
+        Map<String, List<Player>> gameTypes = new HashMap<>();
+        gameTypes.put("1", Arrays.asList(humanVersusHuman));
+        gameTypes.put("2", Arrays.asList(humanVersusComputer));
```
Comment (Maikon) [inline src/main/game/Language/GameFactory.java:26]: I think you can tidy up this a bit by extracting the full values into variables:
```
List<Player> humanVersusHuman = Arrays.asList(new Human(inputOutput), new Human(inputOutput));
List<Player> humanVersusComputer = Arrays.asList(new Human(inputOutput), new Computer());

Map<String, List<Player>> gameTypes = new HashMap<>();
gameTypes.put("1", humanVersusHuman);
gameTypes.put("2", humanVersusComputer);
```

File: src/main/game/Player/Computer.java [hunk 2]
```diff
@@ -13,7 +13,7 @@ public static String computerMove() {
         return moves[randomMove];
     }
 
-    public static Moves playMove() {
+    public Moves playMove() {
```
Comment (Maikon) [inline src/main/game/Player/Computer.java:16]: 👍

File: src/test/game/RulesTest.java [hunk 3]
```diff
@@ -29,51 +30,39 @@ public void setUp() {
...
     public void playerOneWinsWhenPlayerOneIsScissorsAndWinningMoveIsScissors() {
-        Rules rules = new Rules(moves);
-
         assertEquals(Result.PLAYER_ONE_WINS, rules.scoreGame(SCISSORS, SCISSORS));
     }
 
     @Test
     public void playerOneWinsWhenPlayerTwoIsScissorsAndWinningMoveIsScissors() {
-        Rules rules = new Rules(moves);
-
```
Comment (Maikon) [inline src/test/game/RulesTest.java:76]: Nice cleanup on all the tests 👍

File: src/main/game/Language/GameFactory.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+public class GameFactory {
+    private final Ui inputOutput;
+
+    public GameFactory(Ui inputOutput) {
+        this.inputOutput = inputOutput;
+    }
+
+    public List<Player> getPlayerTypes(String playerTypes) {
+        Player[] humanVersusHuman = {new Human(inputOutput), new Human(inputOutput)};
```
Comment (RabeaGleissner) [inline src/main/game/Language/GameFactory.java:21]: I would call it `Players` because it's a list of many.

**Pull request merged**